File: solutions/python/problem_2.py
def main():

    n = 4000000

    f1 = 1
    f2 = 1
    fn = f1 + f2
    total = 0

    while f2 <= n:
        if fn % 2 == 0:
            total += fn

        f1 = f2
        f2 = fn
        fn = f1 + f2

    return total

if __name__ == "__main__":
    print(main())

File: solutions/csharp/Problem2.cs
namespace ProjectEuler
{
    class Problem2 : Problem
    {
        public override int Solve()
        {
            int n = 4000000;

            int f1 = 1;
            int f2 = 1;
            int fn = f1 + f2;
            int total = 0;

            while (f2 <= n)
            {
                if (fn % 2 == 0)
                {
                    total += fn;
                }
                f1 = f2;
                f2 = fn;
                fn = f1 + f2;
            }
            return total;
        }
    }
}